### PR TITLE
Fix manual download instructions

### PR DESCRIPTION
**Pull request by patrickvonplaten**
This PR replaces the static `DatasetBulider` variable `MANUAL_DOWNLOAD_INSTRUCTIONS` by a property function `manual_download_instructions()`. 

Some datasets like XTREME and all WMT need the manual data dir only for a small fraction of the possible configs.

After some brainstorming with @mariamabarham and @lhoestq, we came to the conclusion that having a property function `manual_download_instructions()` gives us more flexibility to decide on a per config basis in the dataset builder if manual download instructions are needed.

Also this PR should unblock solves a bug with `wmt16 - ro-en` 
@sshleifer from this branch you should be able to succesfully run

```python 
import nlp 
ds = nlp.load_dataset('./datasets/wmt16', 'ro-en')
```

and once this PR is merged S3 should be synched so that 

```python
import nlp
ds = nlp.load_dataset("wmt16", "ro-en")
```

works as well.

**Important**: Since `MANUAL_DOWNLOAD_INSTRUCTIONS` was not really exposed to the user, this PR should not be a problem regarding backward compatibility.

### CODE REVIEW

File: datasets/newsroom/newsroom.py [hunk 1]
```diff
@@ -110,7 +113,7 @@ def _split_generators(self, dl_manager):
         if not os.path.exists(data_dir):
             raise FileNotFoundError(
                 "{} does not exist. Make sure you insert a manual dir via `nlp.load('newsroom', data_dir=...)` that includes files unzipped from the reclor zip. Manual download instructions: {}".format(
-                    data_dir, self.MANUAL_DOWNLOAD_INSTRUCTIONS
+                    data_dir, self.manual_download_instructions
```
Comment (lhoestq) [inline datasets/newsroom/newsroom.py:116]:
```suggestion
                    data_dir, self.manual_download_instructions()
```

File: datasets/c4/c4.py [hunk 2]
```diff
@@ -156,6 +147,15 @@ class C4(nlp.BeamBasedBuilder):
         ),
     ]
 
+    def manual_download_instructions(self):
+        return """\
+    For the WebText-like config, you must manually download 'OpenWebText.zip'
+    (from https://mega.nz/#F!EZZD0YwJ!9_PlEQzdMVLaNdKv_ICNVQ) and the Common Crawl
+    WET files from August 2018 to July 2019
+    (https://commoncrawl.org/the-data/get-started/) and place them in the
+    `manual_dir`.
```
Comment (lhoestq) [inline datasets/c4/c4.py:156]: shall we replace mentions of `manual_dir` with `data_dir` ?

File: datasets/wikihow/wikihow.py [hunk 3]
```diff
@@ -107,6 +96,19 @@ class Wikihow(nlp.GeneratorBasedBuilder):
...
+    def manual_download_instructions(self):
+        return """\
+  You need to manually download two wikihow files. An overview of which files to download can be seen at https://github.com/mahnazkoupaee/WikiHow-Dataset.
+  You need to download the following two files manually:
+    1) https://ucsb.app.box.com/s/ap23l8gafpezf4tq3wapr6u8241zz358 and save the file under <path/to/folder>/wikihowAll.csv
+    2) https://ucsb.app.box.com/s/7yq601ijl1lzvlfu4rjdbbxforzd2oag and save the file under <path/to/folder>/wikihowSep.csv
+
+  The <path/to/folder> can e.g. be "~/manual_wikihow_data".
+
+  Wikihow can then be loaded using the following command `nlp.load("wikihow", data_file="<path/to/folder>")`.
```
Comment (lhoestq) [inline datasets/wikihow/wikihow.py:109]:
```suggestion
  Wikihow can then be loaded using the following command `nlp.load("wikihow", data_dir="<path/to/folder>")`.
```
It should be `data_dir` ?

File: datasets/wmt14/wmt14.py [hunk 4]
```diff
@@ -55,6 +55,12 @@ class Wmt14(Wmt):
         for l1, l2 in _LANGUAGE_PAIRS
     ]
 
+    @property
+    def manual_download_instructions(self):
+        if self.config.language_pair[1] in ["cs", "hi", "ru"]:
+            return "Please download the data manually as explained. TODO(PVP)"
```
Comment (lhoestq) [inline datasets/wmt14/wmt14.py:61]: xD

File: datasets/xtreme/xtreme.py [hunk 5]
```diff
@@ -436,6 +431,15 @@ class Xtreme(nlp.GeneratorBasedBuilder):
         for name in _NAMES
     ]
 
+    @property
+    def manual_download_instructions(self):
+        if self.config.name.startswith("PAN-X"):
+            return """\
+             You need to manually download the AmazonPhotos.zip file on Amazon Cloud Drive
+             (https://www.amazon.com/clouddrive/share/d3KGCRCIYwhKJF0H3eWA26hjg2ZCRhjpEQtDL70FSBN) and save the file under <path/to/folder>AmazonPhotos.zip
```
Comment (lhoestq) [inline datasets/xtreme/xtreme.py:439]: maybe mention `data_dir` like in the other datasets

File: src/nlp/builder.py [hunk 6]
```diff
@@ -160,8 +159,8 @@ def __init__(
                 os.rmdir(self._cache_dir)
 
     @property
-    def does_require_manual_download(self):
-        return hasattr(self, "MANUAL_DOWNLOAD_INSTRUCTIONS") and (self.MANUAL_DOWNLOAD_INSTRUCTIONS is not None)
+    def manual_download_instructions(self) -> Optional[str]:
```
Comment (lhoestq) [inline src/nlp/builder.py:162]: Maybe add some docs with example to explain what a user that wants to create a script has to write in this function (cf what's done for _split_generators for example) ?

**Comment by sshleifer** [comment]
Verified that this works, thanks!

**Comment by sshleifer** [comment]
But I get
```python
ConnectionError: Couldn't reach https://s3.amazonaws.com/datasets.huggingface.co/nlp/datasets/./datasets/wmt16/wmt16.py
```
When I try from jupyter on brutasse or my mac.
Both machines can run
```bash
aws s3 ls s3://datasets.huggingface.co/nlp/datasets/wmt16/
```
but it seems one must be in the nlp directory to run the command?

(I ran `pip install -e . ` on this branch in both situations.)

**Comment by julien-c** [comment]
`https://s3.amazonaws.com/datasets.huggingface.co/nlp/datasets/./datasets/wmt16/wmt16.py` looks very weird.

(Also, S3 is not a file-system, it's a flat key-value store)

File: src/nlp/builder.py (hunk 6 above)
Comment (patrickvonplaten) [reply src/nlp/builder.py:162]: sounds good - will keep this for a future PR

**Comment by patrickvonplaten** [comment]
Good to merge I think @lhoestq

**Comment by lhoestq** [comment]
> But I get
> 
> ```python
> ConnectionError: Couldn't reach https://s3.amazonaws.com/datasets.huggingface.co/nlp/datasets/./datasets/wmt16/wmt16.py
> ```
> 
> When I try from jupyter on brutasse or my mac. (the jupyter server is run from transformers).
> 
> Both machines can run
> 
> ```shell
> aws s3 ls s3://datasets.huggingface.co/nlp/datasets/wmt16/
> ```
> 
> but it seems one must be in the nlp directory to run the command?
> 
> (I ran `pip install -e . ` on this branch in both situations.)

As soon as it is on master, the dataset script wmt16.py will be synced on S3 and you'll be able to do `load_dataset("wmt16")`

**Pull request merged**